kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/impl/ShardingSpherePipelineDataSourceConfiguration.java

```diff
@@ -88,18 +91,15 @@ private String getJdbcUrl(final Map<String, Object> props) {
     
     private void appendJdbcQueryProperties(final String databaseType) {
         Optional<JdbcQueryPropertiesExtension> extension = PipelineTypedSPILoader.findDatabaseTypedService(JdbcQueryPropertiesExtension.class, databaseType);
-        if (!extension.isPresent()) {
+        if (!extension.isPresent() || extension.get().extendQueryProperties(new Properties()).isEmpty()) {
             return;
         }
-        Properties queryProps = extension.get().extendQueryProperties();
-        if (queryProps.isEmpty()) {
-            return;
+        for (Entry<String, Map<String, Object>> entry : rootConfig.getDataSources().entrySet()) {
+            String jdbcUrlKey = entry.getValue().containsKey("url") ? "url" : "jdbcUrl";
+            JdbcUrl jdbcUrl = new StandardJdbcUrlParser().parse(entry.getValue().get(jdbcUrlKey).toString());
+            Properties extendedQueryProperties = extension.get().extendQueryProperties(jdbcUrl.getQueryProperties());
+            entry.getValue().replace(jdbcUrlKey, new JdbcUrlAppender().appendQueryProperties(entry.getValue().get(jdbcUrlKey).toString(), extendedQueryProperties));
         }
-        rootConfig.getDataSources()
-                .forEach((key, value) -> {
-                    String jdbcUrlKey = value.containsKey("url") ? "url" : "jdbcUrl";
-                    value.replace(jdbcUrlKey, new JdbcUrlAppender().appendQueryProperties(value.get(jdbcUrlKey).toString(), queryProps));
-                });
     }
```

Comment:
Looks the new block of code has much more characters, could we just add some new line of code in existing forEach block?